pyuvdata/uvdata.py

```diff
@@ -1583,6 +1583,57 @@ def _key2inds(self, key):
                 raise KeyError('Polarization {pol} not found in data.'.format(pol=key[2]))
         return (blt_ind1, blt_ind2, pol_ind)
 
+
+    def _smart_slicing(self, data, ind1, ind2, indp):
+        """
+        Method for quickly picking out the relevant section of data for get_data or get_flags
+
+        Args:
+            data: 4-dimensional array in the format of self.data_array
+            ind1: list with blt indices for antenna pair (e.g. from self._key2inds)
+            ind2: list with blt indices for conjugate antenna pair. (e.g. from self._key2inds)
+            indp: list with polarization indices (e.g. from self._key2inds)
+
+        Returns:
+            out: copy (or if possible, view) of relevant section of data as an numpy array
```

Comment:
Here's a problem: you also copy the data when you conjugate it. I think the typical user wants the data as quickly as possible: a copy if necessary but a view is always preferable since it's much, much faster. On the other hand, I think the user wants pyuvdata to be smart and conjugate the data when the baseline order is backwards. So it's never clear whether the result is going to be a view or a copy.

If the user wants a copy, they can always call `np.array(uv.get_data(...))` which makes a copy. I'd rather clarify the documentation to warn the user against trying to update the underlying data with `get_data`, etc. We can also include a force_copy option if that's helpful.